.github/workflows/deploy.yml

```
@@ -0,0 +1,89 @@
+name: Build and Deploy
+
+on:
+  push:
+    branches:
+      # TODO change to prod
+      # - prod
+      - feature/github-actions
+
+jobs:
+  build_and_test:
+    name: Build and test
+    runs-on: ubuntu-18.04
```

Wouldn't it be better to collapse the two workflows (`build-and-test.yml` and `deploy.yml` into one and add this here?

```suggestion
    runs-on: ubuntu-18.04
    if: github.ref == 'refs/heads/prod'
```

References:

* https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif
* https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions

---

Actually, more like:

```suggestion
    needs: build_and_test
    if: |
      github.event_name == 'push'
      && github.ref == 'refs/heads/prod'
      && github.repository == 'filiph/egamebook_site'
    runs-on: ubuntu-18.04
```

---

I see what you mean. Let's do that. Although I prefer to keep these things separate for easier extensibility, there is a clear code duplicity which cannot be avoided (now) without moving everything into a single file (cross-workflow artefact sharing is not yet supported). I'll only get rid of test with Node 15 which is not a big deal.